aos_cd_jobs/updater.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+from os import getenv, listdir, rename, walk
+from os.path import join, relpath
+from shutil import rmtree
+
+from git import Repo
+
+from aos_cd_jobs.common import JOBS_DIRECTORY, initialize_repo
+
+def update_branches(repo):
+    for job in list_jobs(repo):
+        if job in repo.branches:
+            branch = repo.branches[job]
+        else:
+            branch = None
+        if branch is not None:
+            branch.delete()
+        create_remote_branch(repo, job)
+
+def list_jobs(repo):
+    jobs = []
+    jobs_directory = join(repo.working_dir, JOBS_DIRECTORY)
+    for (dirpath, _, filenames) in walk(jobs_directory):
+        if 'Jenkinsfile' in filenames:
+            jobs.append(relpath(dirpath, jobs_directory))
+    return jobs
+
+def create_remote_branch(repo, name):
```

Comment:
You previously had a test for this function but have deleted it. The rationale behind my comment previously was as such:

The method as is written has a lot of scope. I understand why you were using behavior validation in your test, and I agree that it is the correct means by which you should be testing this, but the previous test was overly specific. Once the test was committed, it would mean that even the slightest change to the method in question would necessarily cause a test to fail and thereby generate more work. If we try to write this as a "what" method instead of a "why" method, we get something like this (arguments elided, but you get the point):

```py
def create_remote_branch():
    initialize_orphan_branch()
    populate_branch()
    publish_branch()
```

Then, we can write very specific implementation-validation tests for the "how" methods like `publish_branch()`. Even though any one test is still very specific and refactoring any one of the underlying "how" methods still leads to test failure, it's scoped. A small code change leads to a small test change.

I would much rather see this than see us drop the test entirely.